tests/algorithms/test_bc.py

```diff
@@ -268,60 +268,49 @@ def test_that_wrong_training_duration_specification_raises_error(
         cartpole_bc_trainer.train(**duration_args)
 
 
-@pytest.mark.parametrize("no_yield_after_iter", [0, 1, 5])
+# Start at 1 as BC uses up an iteration from getting the first element for type checking
+@pytest.mark.parametrize("no_yield_after_iter", [1, 2, 6])
 def test_that_bc_raises_error_when_data_loader_is_empty(
-    cartpole_venv: vec_env.VecEnv,
-    no_yield_after_iter: bool,
-    custom_logger: logger.HierarchicalLogger,
+    no_yield_after_iter: int,
+    cartpole_bc_trainer: bc.BC,
     cartpole_expert_trajectories: Sequence[types.TrajectoryWithRew],
-    rng: np.random.Generator,
+    custom_logger: logger.HierarchicalLogger,
 ) -> None:
     """Check that we error out if the DataLoader suddenly stops yielding any batches.
 
     At one point, we entered an updateless infinite loop in this edge case.
 
     Args:
-        cartpole_venv: Environment to test in.
         no_yield_after_iter: Data loader stops yielding after this many calls.
-        custom_logger: Where to log to.
+        cartpole_bc_trainer: BC trainer.
         cartpole_expert_trajectories: The expert trajectories to use.
-        rng: Random state to use.
+        custom_logger: Where to log to.
     """
     # GIVEN
-    batch_size = 32
+    batch_size = cartpole_bc_trainer.batch_size
     trans = rollout.flatten_trajectories(cartpole_expert_trajectories)
     dummy_yield_value = dataclasses.asdict(trans[:batch_size])
 
     class DataLoaderThatFailsOnNthIter:
         """A dummy DataLoader stops to yield after a number of calls to `__iter__`."""
 
-        def __init__(self, no_yield_after_iter: int):
-            """Builds dummy data loader.
-
-            Args:
-                no_yield_after_iter: `__iter__` will raise `StopIteration` after
-                    this many calls.
-            """
-            self.iter_count = 0
-            self.no_yield_after_iter = no_yield_after_iter
+        iter_count = 0
```

Comment:
Nit: Using a class rather than instance variable here is a bit odd. Doesn't matter right now as we only ever have a single instance of the class. But it could be a gotcha if we change the test later, so I lean slightly towards keeping it in `__init__` even though it's more code?

Reply:
Fixed this.